### ISSUE TITLE
Auto-equip ammo broken

### ISSUE THREAD

**Issue by NoImageAvailable** [NoImageAvailable/CombatExtended#120]
Auto-equip ammo option is currently broken. JobGiver_TakeAndEquip returns an invalid cast error and reportedly ammo and melee weapons aren't being picked up. Should probably give that whole code section a pass to make sure everything is working correctly with the new changes.

**Comment by ProfoundDarkness** [comment]
I've identified the problem area for the invalid cast exception... it's where the jobgiver tries to use the inventory cache of ranged weapons instead of inspecting the inventory to pick out the apt items...

Unable to replicate the other issues as yet.  Melee weapons got picked up as did ammo (pistol and knife).

My test loadout:
Pistol
2 clips of pistol ammo (14)
Knife

The Pistol already had a clip in it so pawn fetched 7 rounds.

**Comment by NoImageAvailable** [comment]
Still not fixed. Steps to replicate:
- Go into mod settings, enable auto-equip ammo option
- Start new game
- Have colonist equip pistol

Produces the following error
```
Exception in RimWorld.ThinkNode_ConditionalCanDoConstantThinkTreeJobNow TryIssueJobPackage: System.InvalidCastException: Cannot cast from source type to destination type.
  at (wrapper managed-to-native) object:__icall_wrapper_mono_object_castclass (object,intptr)
  at System.Linq.Enumerable+<CreateCastIterator>c__Iterator0`1[Verse.ThingDef].MoveNext () [0x00000] in <filename unknown>:0 
  at System.Linq.Enumerable+<CreateSelectIterator>c__Iterator10`2[Verse.ThingDef,Verse.ThingDef].MoveNext () [0x00000] in <filename unknown>:0 
  at System.Collections.Generic.List`1[Verse.ThingDef].AddEnumerable (IEnumerable`1 enumerable) [0x00000] in <filename unknown>:0 
  at System.Collections.Generic.List`1[Verse.ThingDef]..ctor (IEnumerable`1 collection) [0x00000] in <filename unknown>:0 
  at System.Linq.Enumerable.ToList[ThingDef] (IEnumerable`1 source) [0x00000] in <filename unknown>:0 
  at CombatExtended.JobGiver_TakeAndEquip.TryGiveJob (Verse.Pawn pawn) [0x00000] in <filename unknown>:0 
  at Verse.AI.ThinkNode_JobGiver.TryIssueJobPackage (Verse.Pawn pawn) [0x00000] in <filename unknown>:0 
  at Verse.AI.ThinkNode_Priority.TryIssueJobPackage (Verse.Pawn pawn) [0x00000] in <filename unknown>:0 
Verse.Log:Error(String)
Verse.AI.ThinkNode_Priority:TryIssueJobPackage(Pawn)
RimWorld.ThinkNode_Conditional:TryIssueJobPackage(Pawn)
Verse.AI.ThinkNode_Priority:TryIssueJobPackage(Pawn)
Verse.AI.Pawn_JobTracker:DetermineNextConstantThinkTreeJob()
Verse.AI.Pawn_JobTracker:DetermineNextJob(ThinkTreeDef&)
Verse.AI.Pawn_JobTracker:TryFindAndStartJob()
Verse.AI.Pawn_JobTracker:EndCurrentJob(JobCondition, Boolean)
Verse.AI.Pawn_JobTracker:JobTrackerTick()
Verse.Pawn:Tick()
Verse.TickList:Tick()
Verse.TickManager:DoSingleTick()
Verse.TickManager:TickManagerUpdate()
Verse.Game:Update()
Verse.Root_Play:Update()
```

**Comment by NoImageAvailable** [comment]
Confirmed fixed with #135

**Comment by ProfoundDarkness** [comment]
This appears fixed... replication no longer fails and pawn correctly picks up the ammo for the pistol.
